lib/kafo/kafo_configure.rb

```diff
@@ -189,14 +189,18 @@ def help
     def self.help(*args)
       kafo          = args.pop
       builder_class = kafo.full_help? ? HelpBuilders::Advanced : HelpBuilders::Basic
-      args.push builder_class.new(kafo.params)
+      args.push builder_class.new(kafo.params, self.app_options, advanced: kafo.full_help?)
       super(*args)
     end
 
     def self.app_option(*args, &block)
-      self.app_options ||= []
-      self.app_options.push self.option(*args, &block)
-      self.app_options.last
+      self.app_options ||= {}
+      option = self.option(*args, &block)
+
+      advanced = args[3] ? args[3][:advanced] : false
```

Comment:
I never like these indexes. What is `3`? Isn't it cleaner to introduce `**kwargs` as a method parameter? Then advanced ends up in a that hash.